src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java

```diff
@@ -38,4 +61,172 @@ public void waitLoadITuoiRecapitiPage() {
+            return this.emailInputField.isDisplayed();
+    }
+
+    public void insertEmail(String email){
+        this.getWebDriverWait(30).until(ExpectedConditions.visibilityOf(this.emailInputField));
+        this.emailInputField.sendKeys(email);
+    }
+
+    public void clickAvvisamiViaEmail(){
+        this.avvisamiViaEmailButton.click();
+    }
+
+    public boolean avvisamiViaEmailIsDisabled(){
+        try{
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOf(this.avvisamiViaEmailButton));
+            return Boolean.parseBoolean(this.avvisamiViaEmailButton.getAttribute("disabled"));
+        }catch (NoSuchElementException | TimeoutException e){
+            return false;
+        }
+    }
+
+    public String getEmailErrorMessage(){
+        By errorBy = By.id("email-helper-text");
+        WebElement errorMessage = driver.findElement(errorBy);
+        this.getWebDriverWait(30).until(ExpectedConditions.visibilityOf(errorMessage));
+        return errorMessage.getText();
+    }
+
+    public boolean verificaPopupConfirmaEmail(){
+        By hoCapitoCheckboxBy = By.xpath("//span[contains(text(),'Ho capito')]/preceding-sibling::span/input");
+        return this.driver.findElement(hoCapitoCheckboxBy).isSelected();
+    }
+
+    public void clickHoCapitoCheckBoxPopup(){
+        By hoCapitoCheckboxBy = By.xpath("//span[contains(text(),'Ho capito')]/preceding-sibling::span/input");
+        WebElement hoCapitoCheckBox = this.driver.findElement(hoCapitoCheckboxBy);
+        hoCapitoCheckBox.click();
+    }
+
+    public void confirmaEmailPopup(){
+        By popupConfirmaButtonBy = By.xpath("//button[@data-testid='disclaimer-confirm-button']");
+        this.getWebDriverWait(30).until(ExpectedConditions.elementToBeClickable(popupConfirmaButtonBy));
+        this.driver.findElement(popupConfirmaButtonBy).click();
+    }
+
+    public void insertTelephoneNumber(String phoneNumber){
+        this.getWebDriverWait(30).until(ExpectedConditions.visibilityOf(this.phoneNumInputField));
+        this.phoneNumInputField.sendKeys(phoneNumber);
+    }
+
+    public void clickAvvisamiViaSMS(){
+        this.getWebDriverWait(30).until(ExpectedConditions.visibilityOf(this.avvisamiViaSMSButton));
+        this.js().executeScript("arguments[0].click()",this.avvisamiViaSMSButton);
+    }
+
+    public String getPhoneErrorMessage(){
+        By errorBy = By.id("phone-helper-text");
+        WebElement errorMessage = driver.findElement(errorBy);
+        this.getWebDriverWait(30).until(ExpectedConditions.visibilityOf(errorMessage));
+        return errorMessage.getText();
+    }
+
+    public boolean avvisamiViaSMSIsDisabled(){
+        try{
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOf(this.avvisamiViaSMSButton));
+            return Boolean.parseBoolean(this.avvisamiViaSMSButton.getAttribute("disabled"));
+        }catch (NoSuchElementException | TimeoutException e){
+            return false;
```

Comment:
as before

Reply:
Logger already present in the step that calls this method